tests/test_ui_mpldrawer.py

```diff
@@ -0,0 +1,342 @@
+"""Tests for Qibo matplotlib drawer"""
+
+import matplotlib.pyplot
+import numpy as np
+import pytest
+
+from qibo import Circuit, callbacks, gates
+from qibo.models import QFT
+from qibo.ui.drawer_utils import FusedEndGateBarrier, FusedStartGateBarrier
+from qibo.ui.mpldrawer import (
+    _make_cluster_gates,
+    _plot_params,
+    _plot_quantum_circuit,
+    _process_gates,
+    _render_label,
+    plot_circuit,
+)
+
+
+# defining a dummy circuit
+def circuit(nqubits=2):
+    c = Circuit(nqubits)
+    c.add(gates.H(0))
+    c.add(gates.CNOT(0, 1))
+    c.add(gates.M(0))
+    c.add(gates.M(1))
+    return c
+
+
+@pytest.mark.parametrize("nqubits", [2, 3])
+def test_plot_circuit(nqubits):
+    """Test for main plot function"""
+    circ = circuit(nqubits)
+    ax, _ = plot_circuit(circ)
+    assert ax != None

```

Comment:
Take this as a general comment, valid also for the following assertions in the code.

It is the first time I need to test a routine which involves plots, so I am investigating on which can be the best way of doing it. I don't think just checking the axes of our Matplotlib object are different from `None` would be enough. Probably, it would be better to define a strategy to check that also the output is what we expect. 
One possible way of doing this could be to generate some reference plots and constructs some correspondent hash codes. This can be done properly selecting the Matplotlib backend (so that we don't have problem with the rendering of the images adopting different systems) and using the `hashlib` library. 

Once we have some reference hashes (I think just a few plots can be enough to test the whole drawer), we can regenerate the plots in the tests and check if the generated hashes correspond with the target ones. 

Do you have some suggestions/ideas @alecandido @scarrazza?